autotx/agents/TokenResearchAgent.py

```diff
@@ -0,0 +1,173 @@
+import json
+from textwrap import dedent
+from typing import Callable
+from crewai import Agent
+from autotx.AutoTx import AutoTx
+from autotx.auto_tx_agent import AutoTxAgent
+from autotx.utils.coingecko.api import CoingeckoApi
+from gnosis.eth import EthereumClient
+from crewai_tools import BaseTool
+
+coingecko = CoingeckoApi()
+
+
+class TokenSymbolToTokenId(BaseTool):
+    name: str = "token_symbol_to_token_id"
+    description: str = dedent(
+        """
+        Fetch tokens list from coingecko,
```

Comment:
I don't think we should make reference to `coingecko` in the agent prompts, that's an implementation detail of the function.